packages/x-components/src/x-modules/experience-controls/store/actions/fetch-and-save-controls.action.ts

```diff
@@ -0,0 +1,31 @@
+import { createFetchAndSaveActions } from '../../../../store/utils/fetch-and-save-action.utils';
+import { ExperienceControlsActionContext } from '../types';
+
+const { fetchAndSave, cancelPrevious } = createFetchAndSaveActions<
+  ExperienceControlsActionContext,
+  //TODO: change types when the adapter is updated
+  any,
+  any
+>({
+  fetch({ dispatch }, request) {
+    return dispatch('fetchExperienceControlsResponse', request);
+  },
+  onSuccess({ commit }, controls) {
+    commit('setControls', controls);
+    commit('setEvents', {});
+  }
+});
```

Comment:
This still doesn't take into account that the object returned by the success of the fetch will be similar to `{ controls, events }`

```suggestion
  onSuccess({ commit }, experienceControlsResponse) {
    commit('setControls', experienceControlsResponse.controls);
    commit('setEvents', experienceControlsResponse.events);
  }
});
```